src/rebuild/srv.c

```diff
@@ -2506,14 +2514,16 @@ rebuild_tgt_prepare(crt_rpc_t *rpc, struct rebuild_tgt_pool_tracker **p_rpt)
 	rpt->rt_pool = pool; /* pin it */
 	ABT_mutex_unlock(rpt->rt_lock);
 
-	rpt_get(rpt);
-	d_list_add(&rpt->rt_list, &rebuild_gst.rg_tgt_tracker_list);
 	*p_rpt = rpt;
 out:
 	if (rc) {
-		if (rpt)
+		if (rpt) {
 			rpt_put(rpt);
-
+			if (!d_list_empty(&rpt->rt_list)) {
```

Comment:
This check better put before above rpt_put() to avoid accessing freed memory?

Reply:
Hmm, actually if rpt is created successfully, then rpt must be in the list anyway. Though yes, it is better to put after rt_list indeed. thanks.